lib/AST/ASTPrinter.cpp

```diff
@@ -3749,19 +3757,37 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
   }
 
   void visitParentType(Type T) {
+    /// Don't print the parent type if it's being printed in that type context.
+    if (Options.TransformContext) {
+      if (auto currentType = Options.TransformContext->getBaseType()) {
+        auto printingType = T;
+        if (currentType->hasArchetype())
+          currentType = currentType->mapTypeOutOfContext();
+
+        if (auto errorTy = printingType->getAs<ErrorType>())
```

Comment:
When does it have an error type?

Reply:
```swift
protocol P {
 associatedtype A
 func foo(x: A)
}
struct S: P {
 func #^COMPLETE^#
}
````
When printing `foo(x:)` method in this completion, the parameter type is `S.A` where `S` is `ErrorType` with original type:
```
(dependent_member_type assoc_type=MyModule.(file).P.A@test.swift:2:17
  (base=error_type
    (original_type=struct_type decl=MyModule.(file).S@test.swift:5:8)))
```